examples/custom_distance/example_custom_distance.cpp

```diff
@@ -0,0 +1,140 @@
+/****************************************************************************
+ * Copyright (c) 2025, ArborX authors                                       *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+#include <vector>
+
+// Define custom type that will be used for distance calculations
+struct CustomPoint
+{
+  float x;
+  float y;
+
+  KOKKOS_FUNCTION auto operator[](int i) const
+  {
+    KOKKOS_ASSERT(i == 0 || i == 1);
+    return (i == 0 ? x : y);
+  }
+  KOKKOS_FUNCTION auto &operator[](int i)
+  {
+    KOKKOS_ASSERT(i == 0 || i == 1);
+    return (i == 0 ? x : y);
+  }
+};
+
+template <>
+struct ArborX::GeometryTraits::dimension<CustomPoint>
+{
+  static constexpr int value = 2;
+};
+template <>
+struct ArborX::GeometryTraits::coordinate_type<CustomPoint>
+{
+  using type = float;
+};
+template <>
+struct ArborX::GeometryTraits::tag<CustomPoint>
+{
+  using type = ArborX::GeometryTraits::PointTag;
+};
+
+// Provide the required centroid function for the custom data type
+KOKKOS_FUNCTION auto returnCentroid(CustomPoint const &point) { return point; }
```

Comment:
How comes we don't default that implementation?
(not blocking but we should consider fixing)

Reply:
We do. I removed it. It was from the time when `CustomPoint` did not have `operator[]`, and `returnCentroid` returned `ArborX::Point` and not `CustomPoint`.